tests/test_table_row_classification.py

```diff
@@ -36,3 +36,50 @@ def test_table_row_classification(tmpdir, feature_table):
 
     assert result.dtype == np.uint32
     assert np.allclose(ground_truth, result)
+
+@pytest.mark.parametrize("feature_table", [feature_table_dict, feature_table_data_frame])
+def test_with_nans(tmpdir, feature_table):
+
+    feature_table['surface_area'][4] = np.nan
+
+    print(feature_table)
+
+    annotation = np.array([1, 1, 2, 2, 0])
+    reference = np.array([1, 1, 2, 2, 0])
+
+    import apoc
+    opencl_file = tmpdir.join("test.cl")
+    oc = apoc.TableRowClassifier(opencl_file)
+    oc.train(
+        feature_table=feature_table,
+        ground_truth=annotation,
+        continue_training=False
+    )
+    result = oc.predict(feature_table, return_numpy=True)
+
+    np.allclose(result, reference)
+
+@pytest.mark.parametrize("feature_table", [feature_table_dict, feature_table_data_frame])
+def test_with_nans_and_missing_annotations(tmpdir, feature_table):
```

Comment:
It might be nice to add a unit test for the `TableRowClassifier._prepare_feature_table()` method to make sure the nans and missing annotations are actually dropped.